src/core/Table/columns/actionColumn.tsx

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { HeaderProps } from 'react-table';
+import { Checkbox } from '../../Checkbox';
+import SvgColumnManager from '@itwin/itwinui-icons-react/cjs/icons/ColumnManager';
+import { DropdownMenu } from '../../DropdownMenu';
+import { IconButton } from '../../Buttons/IconButton';
+import { MenuItem } from '../../Menu';
+import { TABLE_RESIZE_START_ACTION } from '../actionHandlers';
+import { EXPANDER_CELL_ID, SELECTION_CELL_ID } from '../hooks';
+
+const ACTION_CELL_ID = 'iui-table-action';
+
+export const ActionColumn = <T extends Record<string, unknown>>({
+  hasColumnManager = true,
+}) => {
+  return {
+    id: ACTION_CELL_ID,
+    disableResizing: true,
+    disableGroupBy: true,
+    minWidth: 48,
+    width: 48,
+    maxWidth: 48,
+    columnClassName: 'iui-slot',
+    cellClassName: 'iui-slot',
+    disableReordering: true,
+    Header: ({ allColumns, dispatch }: HeaderProps<T>) => {
+      const [isOpen, setIsOpen] = React.useState(false);
+      if (!hasColumnManager) {
+        return null;
+      } else {
+        const headerCheckBoxes = () =>
+          allColumns
+            //Filters out any default columns made such as selection and expansion
+            .filter(
+              ({ id }) =>
+                ![SELECTION_CELL_ID, EXPANDER_CELL_ID, ACTION_CELL_ID].includes(
```

Comment:
minor, but could have generated this array as const above, instead of doing that on every column item